docs/source/guides/guides_index.rst

```diff
@@ -12,3 +12,4 @@ The guides below provide more detail on the functionality of Woodwork.
    using_woodwork_with_dask_and_koalas
    custom_types_and_type_inference
    saving_and_loading_dataframes
+   logical_types_and_semantic_tags
```

Comment:
ah, that might be because there are too many headings with the singe `#` size header. I'll fix that

Reply:
We should also move this higher up. Perhaps the 2nd bullet?

Reply:
Makes sense to me to move up.

I also think the `Understanding Types and Tags` guide could benefit from being renamed since it provides an overview of Woodwork types but mostly spends its time explaining how to set and update these types. I'm thinking:
- Understanding Types and Tags --> Working with Types and Tags
- this guide --> Understanding Logical Types and Semantic Tags

Reply:
Making changes that end up with this as the list of guides
![image](https://user-images.githubusercontent.com/64278226/127684386-cc2e7709-65fc-4079-ad9b-1f0d4fd9b6ef.png)